reversed_linkedlist.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import print_function
+
+class Node(object):
+    """
+    Node implementation
+    """
+    def __init__(self, payload=None, next_node=None):
+        self.payload = payload
+        self.next_node = next_node
+
+    def get_payload(self):
+        return self.payload
+
+    def get_next(self):
+        return self.next_node
+
+    def has_next(self):
+        return bool(self.next_node)
+
+    def set_next(self, new_next):
+        self.next_node = new_next
+        return self.next_node
+
+    def __str__(self):
+        return self.payload
+
+class LinkedList(object):
+    """
+    Singly-linked list implementation
+    """
+    def __init__(self, head=None):
+        self.head = head
+
+    def get_head(self):
+        return self.head
+
+    def get_tail(self):
+        current = self.head
```

Comment:
It's just a convenient method for me. One can always easily get to the tail of a linked list. I ended up using it only in the `append` method (which was not relevant at all).